packages/muelu/src/Transfers/Matrix-Free/MueLu_MatrixFreeTentativeP_kokkos_decl.hpp

```diff
@@ -0,0 +1,141 @@
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact
+//                    Jonathan Hu       (jhu@sandia.gov)
+//                    Andrey Prokopenko (aprokop@sandia.gov)
+//                    Ray Tuminaro      (rstumin@sandia.gov)
+//
+// ***********************************************************************
+//
+// @HEADER
+#ifndef MUELU_MATRIXFREETENTATIVEP_KOKKOS_DECL_HPP
+#define MUELU_MATRIXFREETENTATIVEP_KOKKOS_DECL_HPP
+
+#include "MueLu_ConfigDefs.hpp"
+#ifdef HAVE_MUELU_KOKKOS_REFACTOR
+
+#include "MueLu_MatrixFreeTentativeP_kokkos_fwd.hpp"
+
+#include "Xpetra_Map.hpp"
+#include <KokkosCompat_ClassicNodeAPI_Wrapper.hpp>
+
+#include "Teuchos_ScalarTraits.hpp"
+
+#include "MueLu_Aggregates_kokkos_fwd.hpp"
+#include "MueLu_PerfUtils_fwd.hpp"
+#include "Xpetra_Operator.hpp"
+#include "MueLu_Utilities_kokkos_fwd.hpp"
+
+namespace MueLu {
+
+  /*!
+    @class MatrixFreeTentativeP class.
+    @brief Matrix-free tentative restrictor operator.
+  */
+  // template <class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
+  // class MatrixFreeTentativeP_kokkos;
+
+  template <class Scalar, class LocalOrdinal, class GlobalOrdinal, class DeviceType>
+  class MatrixFreeTentativeP_kokkos<Scalar, LocalOrdinal, GlobalOrdinal, Kokkos::Compat::KokkosDeviceWrapperNode<DeviceType>> : public Xpetra::Operator<Scalar,LocalOrdinal,GlobalOrdinal,Kokkos::Compat::KokkosDeviceWrapperNode<DeviceType>> {
+  public:
+    typedef LocalOrdinal                                             local_ordinal_type;
+    typedef GlobalOrdinal                                            global_ordinal_type;
+    typedef typename DeviceType::execution_space                     execution_space;
+    typedef Kokkos::RangePolicy<local_ordinal_type, execution_space> range_type;
+    typedef Kokkos::Compat::KokkosDeviceWrapperNode<DeviceType>      node_type;
+    typedef typename Teuchos::ScalarTraits<Scalar>::coordinateType   real_type;
+
+  private:
+    // For compatibility
+    typedef node_type                                           Node;
+#undef MUELU_MATRIXFREETENTATIVEP_KOKKOS_SHORT
+#include "MueLu_UseShortNames.hpp"
+
+  public:
+    //! @name Constructors/Destructors.
+    //@{
+
+    //! Constructor
+    MatrixFreeTentativeP_kokkos(Teuchos::RCP<const Map> coarse_map, Teuchos::RCP<const Map> fine_map, Teuchos::RCP<const Aggregates_kokkos> aggregates)
+    : fine_map_(fine_map),
+      coarse_map_(coarse_map),
+      aggregates_(aggregates)
+    { }
+
+    //! Destructor.
+    ~MatrixFreeTentativeP_kokkos() { }
```

Comment:
I think you can use `= default` instead of an empty one.

Reply:
I think @mayrmt had a similar comment on #10833. I've been copying what is already in MueLu, and it seems like we tend to be more pedantic with writing constructors/destructors out, even if they're the default behavior. I don't know how consistently this is being done across MueLu, however.

Reply:
Done!